app/src/main/java/com/example/easyupipayment/MainActivity.java

```diff
@@ -27,64 +31,122 @@ protected void onCreate(Bundle savedInstanceState) {
 
         //Initialize Components
         imageView = findViewById(R.id.imageView);
-        statusView = findViewById(R.id.textView_status);
+        amount = findViewById(R.id.amount);
+        upiid = findViewById(R.id.UpiId);
+        note = findViewById(R.id.Note);
+        payee = findViewById(R.id.PayeeName);
         payButton = findViewById(R.id.button_pay);
 
-        //Create instance of EasyUpiPayment
-        final EasyUpiPayment easyUpiPayment = new EasyUpiPayment.Builder()
-                .with(this)
-                .setPayeeVpa("example@vpa")
-                .setPayeeName("PAYEE_NAME")
-                .setTransactionId("TRANSACTION_ID")
-                .setTransactionRefId("TRANSACTION_REF_ID")
-                .setDescription("DESCRIPTION_OR_SHORT_NOTE")
-                .setAmount("AMOUNT IN XX.XX DECIMAL FORMAT")
-                .build();
-
-        //Register Listener for Events
-        easyUpiPayment.setPaymentStatusListener(this);
-
         //Proceed for Payment on click
         payButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                easyUpiPayment.startPayment();
+                final String AMOUNT = amount.getText().toString();
+                final String UPI = upiid.getText().toString();
+                final String NOTE = note.getText().toString();
+                final String PayeeName = payee.getText().toString();
+
+                //Create instance of EasyUpiPayment
+                if(AMOUNT.isEmpty() || UPI.isEmpty() || NOTE.isEmpty() || PayeeName.isEmpty()){
+                    Toast.makeText(MainActivity.this, "All fields are mandatory.", Toast.LENGTH_SHORT).show();
+                }else {
+                    double amt = Double.parseDouble(AMOUNT);
+                    final EasyUpiPayment easyUpiPayment = new EasyUpiPayment.Builder()
+                            .with(MainActivity.this)
+                            .setPayeeVpa(UPI)
+                            .setDescription(NOTE)
+                            .setAmount(String.valueOf(amt))
+                            .setPayeeName(PayeeName)
+                            .setTransactionId("TRANSACTION_ID")
+                            .setTransactionRefId("TRANSACTION_REF_ID")
```

Comment:
For demonstration purposes, There should two `EditText`s for Transaction ID and Transaction Ref ID.